uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Helper/NavigationMenuHelper.php

```diff
@@ -114,8 +121,14 @@ private static function relocateWithOffset(
       throw new \Exception($err);
     }
 
+    // we disallow zero offset insertion to make the function clearer,
+    // however this means positive offsets are 1 higher than they should be
+    if ($offset > 0) {
+      $offset--;
```

Comment:
one is subtracted here and then added back again for when offset is positive.
This seems like it will be better represented by an if /else
```php
if($offset>0) {
$insertionIndex += $offset
}
else{
$insertionIndex += ($offset + 1)
}

Reply:
Good point, although since we always add the offset and only add 1 if it's negative I changed it to 

```
    $insertionIndex += $offset;
    if ($offset < 0) {
      $insertionIndex++;
    }
```

Reply:
What if offset is -3?
```php
$insertionIndex += ($offset + 1)
```
This will take care of that scenario.
Also we should watch out for when insertion index becomes negative, what do we do then?